### PR TITLE
Wayland: fixes

### PR DESCRIPTION
**Pull request by Sunderland93**
I'm working on several fixes for Wayland context. DON"T MERGE, before code is reviewed and tested

### CODE REVIEW

**Comment by twinaphex** [comment]
OK, keep in mind though that since owning an Nvidia GPU, I can no longer test Wayland myself, so hopefully you can do the heavylifting there in terms of testing.

I could test it before on a system with an Intel iGPU, but I will have to replace a harddrive in it first since the SSD wore out and I don't have a replacement yet for it.

File: gfx/drivers_context/wayland_ctx.c [hunk 1]
```diff
@@ -571,17 +574,55 @@ static const struct xdg_surface_listener xdg_surface_listener = {
...
+			}
+	}
+	if (width > 0 && height >0) {
+				wl->prev_width = width;
+				wl->prev_height = height;
+				wl->width = width;
+				wl->height = height;
+			}
+			else {
+				wl->width = wl->prev_width;
```
Comment (Sunderland93) [inline gfx/drivers_context/wayland_ctx.c]: Initial configure event is "width 0" and "height 0", so RetroArch window is black, but after resize it drawning fine.

**Comment by Sunderland93** [comment]
Please, review

**Comment by twinaphex** [comment]
Yeah, it's best you don't use any printfs. Always use RARCH_LOG, RARCH_WARN, RARCH_ERR instead if you need to show log notifications.

File: gfx/drivers_context/wayland_ctx.c [hunk 2]
```diff
@@ -1423,9 +1464,9 @@ static bool gfx_ctx_wl_has_focus(void *data)
 
 static bool gfx_ctx_wl_suppress_screensaver(void *data, bool state)
 {
-	(void)data;
-	gfx_ctx_wayland_data_t *wl = (gfx_ctx_wayland_data_t*)data;
-	
+        (void)data;
```
Comment (twinaphex) [inline gfx/drivers_context/wayland_ctx.c]: This is not C89-compliant. Declare variables at the top of a function or code block.

File: gfx/drivers_context/wayland_ctx.c [hunk 3]
```diff
@@ -571,17 +574,55 @@ static const struct xdg_surface_listener xdg_surface_listener = {
...
-    (void)toplevel;
-    
-    wl->width = wl->buffer_scale * width;
-    wl->height = wl->buffer_scale * height;
-    
-    wl->configured = false;
+    printf("Configure event got, width: %d, height: %d\n", width, height);
+    wl->fullscreen = false;
+    wl->maximized = false;
+    enum xdg_toplevel_state *state;
```
Comment (twinaphex) [inline gfx/drivers_context/wayland_ctx.c]: Same thing here. Not C89 compliant. Declare variables at the top of a function or code block.

**Comment by Sunderland93** [comment]
I found workaround for fixing surface resize. I'm not so sure, but it works. @twinaphex can you review again?

**Comment by Sunderland93** [comment]
Overall works on GNOME, Sway, Rootston and Mir without vidible isues.

**Comment by Sunderland93** [comment]
It's ready to merge

**Comment by Sunderland93** [comment]
@bparker06 can you please review and merge this PR? I have done my next patch, and it wait to send.

**Comment by hizzlekizzle** [comment]
@Sunderland93 I spoke with @bparker06 and he suggested that there are still some C98 compatibility and style issues that need to be resolved.

**Comment by Sunderland93** [comment]
I didn't find any C89 compatible issues with `gcc -std=c89 -pedantic` and `gcc -std=c89 -pedantic -Wall -Wextra -Wstrict-prototypes -Wold-style-definition -Wmissing-prototypes -Wmissing-declarations -Wdeclaration-after-statement -Werror=declaration-after-statement`

**Pull request merged**